harness/go/harness/runner.go

```diff
@@ -304,6 +294,24 @@ func (r *Runner) CheckAssertion(result bool) error {
 	return nil
 }
 
+func (r *Runner) RestartWorker() error {
+	r.Worker = worker.New(r.Client, r.RunnerConfig.TaskQueue, r.Feature.WorkerOptions)
```

Comment:
Mentioned in other comment, but should fail if `r.Worker` is already set and have a `StopWorker` unset it.